File: Week 5 - 10. SimpleTabAppCustom/SimpleTabAppKai/FirstViewController.m
//
//  FirstViewController.m
//  SimpleTabAppKai
//
//  Created by  Julio Reyes  on 4/1/15.
//  Copyright (c) 2015 JulioReyesCorps. All rights reserved.
//

#import "FirstViewController.h"

@interface FirstViewController (){
    int counter1;
    int counter2;
}

@property (nonatomic, weak) IBOutlet UITextField *txtCounter1;
@property (nonatomic, weak) IBOutlet UITextField *txtCounter2;

@end

@implementation FirstViewController
- (instancetype)init
{
    self = [super init];
    if (self) {
        self.title = NSLocalizedString(@"First", @"First");
        self.tabBarItem.image = [UIImage imageNamed:@"first"];
        
        
        counter1 = 0;
        counter2 = 0;
        
        [[NSNotificationCenter defaultCenter] addObserver:self
                                                 selector:@selector(receiveNotification:) name:@"Test1" object:nil];
        [[NSNotificationCenter defaultCenter] addObserver:self
                                                 selector:@selector(receiveNotification:) name:@"Test2" object:nil];
    }
    return self;
}
-(void)receiveNotification:(NSNotification *) notification{
    
    NSLog(@"First View Notification Received: %@", [notification name]);
    
    NSDictionary *extraInfo = [notification userInfo];
    
    if ([[notification name] isEqualToString:@"Test1"]) {
        counter1++;
        _txtCounter1.text = [NSString stringWithFormat:@"The Sun's light has burned Moon's gaze.%@: %d",
                            [extraInfo objectForKey:@"button_name"], counter1];
    }
    else{
        counter2++;
        _txtCounter2.text = [NSString stringWithFormat:@"The Moon's darknees has extingushed Sun's flames. %@: %d",
                            [extraInfo objectForKey:@"button_name"], counter2];
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Swift/class 15/SwiftTabAppSample/SwiftTabAppSample/FirstViewController.swift
//
//  FirstViewController.swift
//  SwiftTabAppSample
//
//  Created by Oren Goldberg on 12/10/14.
//  Copyright (c) 2014 Turn To Tech. All rights reserved.
//

import UIKit

class FirstViewController: UIViewController {
    @IBOutlet var txtCounter1: UITextField!
    @IBOutlet var txtCounter2: UITextField!
    var counter1:Int = 0
    var counter2:Int = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "receiveNotification:", name: "Test1", object: nil)
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "receiveNotification:", name: "Test2", object: nil)
        
    }
    
    func receiveNotification(notification:NSNotification) {
        println("First View Notification Received: \(notification.name)")
        var extraInfo:Dictionary = notification.userInfo!
        
        if (notification.name == "Test1") {
            counter1++
            var button_name:AnyObject = extraInfo["button_name"]!
            txtCounter1.text = "\(button_name) \(counter1)"
        } else {
            counter2++
            var button_name:AnyObject = extraInfo["button_name"]!
            txtCounter2.text = "\(button_name) \(counter2)"
        }
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}